R/lt_single2abridged.R

```diff
@@ -0,0 +1,66 @@
+# An abridged life table that is coherent with an input life table by single year of age
+
+#' calculate an abridged life table that is consistent with a life table by single year of age
+#' @description Computes abridged life table columns based on the lx, nLx , and ex values from
+#' a single year life table, in accordance with step 2.2 of the Human Life Table Protocol
+#' https://www.lifetable.de/methodology.pdf. Output abridged life table has same open age group
+#' as input single age life table
+#' @details Similar to \code{lt_abridged()} details, forthcoming
+#' @param lx numeric. Vector of lifetable survivorship at single ages.
+#' @param nLx numeric. Vector of lifetable exposure at single ages.
+#' @param ex numeric. Vector of Age-specific remaining life expectancy at single ages.
+#' @return Abridged lifetable in data.frame with columns
+#' \itemize{
+#'   \item{Age}{integer. Lower bound of abridged age class},
+#'   \item{AgeInt}{integer. Age class widths.}
+#'   \item{nMx}{numeric. Age-specific central death rates.}
+#'   \item{nAx}{numeric. Average time spent in interval by those deceased in interval. }
+#'   \item{nqx}{numeric. Age-specific conditional death probabilities.}
+#'   \item{lx}{numeric. Lifetable survivorship}
+#'   \item{ndx}{numeric. Lifetable deaths distribution.}
+#'   \item{nLx}{numeric. Lifetable exposure.}
+#'   \item{Sx}{numeric. Survivor ratios in uniform 5-year age groups.}
+#'   \item{Tx}{numeric. Lifetable total years left to live above age x.}
+#'   \item{ex}{numeric. Age-specific remaining life expectancy.}
+#' }
+#' @export
+#' 
+lt_single2abridged <- function(lx,
+                               nLx,
```

Comment:
I think only two of these vectors are needed. By identity, we have `ex = Tx/lx`, and `Tx` gives us `Lx`. Any two of these should suffice. One supposes that the whole lifetable is on hand, though, so maybe no need to aim for maximum parsimony?